Code/Tools/AssetProcessor/native/AssetManager/AssetRequestHandler.cpp

```diff
@@ -73,6 +74,36 @@ namespace
     using namespace AzToolsFramework::AssetSystem;
     using namespace AzFramework::AssetSystem;
 
+    //! utility function - splits a string into lines and outputs them to the console at the same time as a trace.
+    void ParseToLines(AZStd::vector<AZStd::string>& lines, const AZStd::string& text)
+    {
+        AzFramework::StringFunc::TokenizeVisitor(
+            text,
+            [&lines](AZStd::string line)
+            {
+                lines.push_back(line);
+                AZ::Debug::Trace::Instance().Output(AssetProcessor::ConsoleChannel, (line + "\n").c_str());
+            },
+            "\n");
+    }
+
+    // generic version of BuildFailure, generally assumes that the failure type is a string.
+    template<typename T> 
+    void BuildFailure(T& failure,  AZStd::vector<AZStd::string>& lines)
```

Comment:
The failure parameter should be `const T&` since `ParseToLines` accepts a `const AZStd::string&`
```suggestion
    void BuildFailure(const T& failure,  AZStd::vector<AZStd::string>& lines)
```